src/server/api.ts

```diff
@@ -280,20 +280,51 @@ export const setupApi = (
     },
   )
 
+  const commonIssueToProjectFieldRuleCreationSchemaKeys = {
```

Comment:
The previous `.when("field"`)` approach didn't work as expected when one field was missing from the object or when I allowed them to be `null` (as they should). This approach works as expected for the all the use cases I've tried.

And the error message ends up pretty good

```json
{
  "_original": { "project_field": "foo", "project_field_value": null },
  "details": [
    {
      "message": "\"project_field\" must be [null]",
      "path": ["project_field"],
      "type": "any.only",
      "context": {
        "valids": [null],
        "label": "project_field",
        "value": "bla",
        "key": "project_field"
      }
    }
  ]
}
```

Reply:
Do you think that a good API approach to force user to list `null` fields, instead of omitting those?

Reply:
> It also feels quite inconsistent with the filter field, which is optional and doesn't allow null

Good that you pointed it out, I fixed that in https://github.com/paritytech/github-issue-sync/pull/17/commits/ed7865cf464db960eb706cbc9eb196585cea2a82

Reply:
> Do you think that a good API approach to force user to list null fields, instead of omitting those

They have different semantics for update. There a missing field means "do not update this field", while `field: null` means "clear this field" (set it to null).

> Does it... require one value to be null, when the other one is null, but doesn't require them to be present?

If one of them is `null`, the other has to be null. Meanwhile, if one of them is required (string), the other has to be, to. For insert both of them should be required while for update they are both either missing (`undefined`), or both `null`, or both `required`. While explaining this I realized we can slightly improve the validation for the insert case, but the update case should stay as-is.

Reply:
> While explaining this I realized we can slightly improve the validation for the insert case, but the update case should stay as-is.

d661df6500fd123c693d0efbe3e226902a12b3cf

Reply:
> They have different semantics for update. There a missing field means "do not update this field", while field: null means "clear this field" (set it to null).

Okay, that makes sense

>  For insert both of them should be required

But why? I really don't think that we need to spell out any optional parameters, when calling the API. Let's default that to null, without forcing users to add it.
If we add another optional parameter, we won't make it mandatory to pass nulls, right?

Reply:
I also feel like this Joi structure is a bit hard to understand, maybe a comment explaining it would be helpful

Reply:
> But why? I really don't think that we need to spell out any optional parameters, when calling the API. Let's default that to null, without forcing users to add it.

I got confused and misunderstood the requirement. Insert/update should not have different semantics for this field, contrary to what I explained above. Reverted in https://github.com/paritytech/github-issue-sync/pull/17/commits/f39ff47fce418465cf2254d7ac84f9c864a550b8.

Reply:
> I also feel like this Joi structure is a bit hard to understand, maybe a comment explaining it would be helpful

https://github.com/paritytech/github-issue-sync/pull/17/commits/ec34678e1e681c1c86f3d5e0b4009f16f477a88e and https://github.com/paritytech/github-issue-sync/pull/17/commits/8a0a4d853485c7a691537abb6b08253c5953fa26